src/main/java/vg/civcraft/mc/civchat2/listeners/CivChat2Listener.java

```diff
@@ -55,18 +57,34 @@ public void onPlayerJoin(PlayerJoinEvent playerJoinEvent) {
 		if (!CivChat2.getInstance().getPluginConfig().getLoginAnnounce()) {
 			playerJoinEvent.setJoinMessage(null);
 		}
-		for (Player p : Bukkit.getOnlinePlayers()){
-			if (settings.getShowJoins(p.getUniqueId())){
+		for (Player p : Bukkit.getOnlinePlayers()) {
+			if (settings.getShowJoins(p.getUniqueId())) {
 				p.sendMessage(playerJoinEvent.getPlayer().getDisplayName() + ChatColor.YELLOW + " has joined the game");
 			}
 		}
+
+		if (CivChat2.getInstance().getPluginConfig().getChatRangeWarn() && !playerJoinEvent.getPlayer().hasPlayedBefore()) {
+			localWarn.add(playerJoinEvent.getPlayer().getUniqueId());
+		}
 	}
 
 	@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
 	public void onPlayerKick(PlayerKickEvent playerKickEvent) {
 		playerKickEvent.setLeaveMessage(null);
 	}
 
+	@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+	public void onGlobalChatEvent(GlobalChatEvent localchat) {
+
+		if (localWarn.contains(localchat.getPlayer().getUniqueId())) {
+			localchat.getPlayer().sendMessage(ChatColor.GOLD
+					+ "You are in local chat! Only players within "
```

Comment:
This implies that there is a (physical) way to not be in local chat. Just don't include that first sentence.